filebeat/docs/how-filebeat-works.asciidoc

```diff
@@ -1,16 +1,25 @@
 [[how-filebeat-works]]
-== How Filebeat works
+== How {beatname_uc} works
 
-In this topic, you learn about the key building blocks of Filebeat and how they work together. Understanding these concepts will help you make informed decisions about configuring Filebeat for specific use cases.
+// REVIWERS: I think we need to modify this topic to be more generic, or we
+// need to explain how different inputs handle the event data differently. It
+// doesn't seem accurate to use the word "file" everywhere as we do here. I
+// tried replacing file with log, but that doesn't work in all situations, so
+// I backed out my changes, and I'm looking for alternatives. Maybe we should
+// make this description more generic, and move the details about reading files
+// to the new topic about the log input. WDYT? I might do that in a separate
+// PR so we can get the refactoring merged.
 
-Filebeat consists of two main components: <<prospector,prospectors>> and <<harvester,harvesters>>. These components work together to tail files and send event data to the output that you specify.
+In this topic, you learn about the key building blocks of {beatname_uc} and how they work together. Understanding these concepts will help you make informed decisions about configuring {beatname_uc} for specific use cases.
+
+{beatname_uc} consists of two main components: <<input,inputs>> and <<harvester,harvesters>>. These components work together to tail files and send event data to the output that you specify.
```

Comment:
TBH I don't like this documentation file. The point is, the presence of a harvester and a particular harvester implementation is 'implementation specific'. A harvester is basically just a worker, allowing for concurrent collection of logs.

I understand the need of this doc, but in the end, we might have inputs and harvesters/workers document for each input type.

Different input types might have different strategies for collecting logs. E.g. for stdin there is no real need for an harvester.

We're also about entangling the harvesters, to have per input type worker implementations. The log harvester is overkill for some use-cases, but re-use 'complicates' the logic.

Reply:
Yah, I agree. I'm going to track this work in a separate issue: https://github.com/elastic/beats/issues/6572